src/hotspot/share/classfile/packageEntry.cpp

```diff
@@ -169,28 +188,26 @@ void PackageEntry::delete_qualified_exports() {
   _qualified_exports = NULL;
 }
 
-PackageEntryTable::PackageEntryTable(int table_size)
-  : Hashtable<Symbol*, mtModule>(table_size, sizeof(PackageEntry))
-{
-}
+PackageEntryTable::PackageEntryTable() { }
 
 PackageEntryTable::~PackageEntryTable() {
-  // Walk through all buckets and all entries in each bucket,
-  // freeing each entry.
-  for (int i = 0; i < table_size(); ++i) {
-    for (PackageEntry* p = bucket(i); p != NULL;) {
-      PackageEntry* to_remove = p;
-      // read next before freeing.
-      p = p->next();
-
-      // Clean out the C heap allocated qualified exports list first before freeing the entry
-      to_remove->delete_qualified_exports();
-      to_remove->name()->decrement_refcount();
-
-      BasicHashtable<mtModule>::free_entry(to_remove);
+  class PackageEntryTableDeleter : public StackObj {
+   public:
+    bool do_entry(const Symbol*& name, PackageEntry*& entry) {
+      if (log_is_enabled(Info, module, unload) || log_is_enabled(Debug, module)) {
+        ResourceMark rm;
+        const char* str = name->as_C_string();
+        log_info(module, unload)("unloading package %s", str);
+        log_debug(module)("PackageEntry: deleting package: %s", str);
```

Comment:
A real nit but to be consistent with ModuleEntryTable::~ModuleEntryTable() maybe change "PackageEntry:" to "PackageEntryTable:"